pkg/noderesourcetopology/cache/overreserve.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	k8scache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesourcetopology/stringify"
+
+	"github.com/k8stopologyawareschedwg/podfingerprint"
+)
+
+type OverReserve struct {
+	lock             sync.Mutex
+	nrts             nrtStore
+	assumedResources map[string]resourceStore // nodeName -> resourceStore
```

Comment:
`resourceStore` is a pod -> v1.ResourceList, but the pod info is not used in UpdateNRT(), right?

If so, maybe we can consider simplifying the structure to make it a `map[string]v1.ResourceList` so as to make UpdateNRT() from a 3-level for loop to a 2-level for loop.

Reply:
OTOH we can get rid of `ContainsPod` and simplify at least a bit